src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -423,14 +493,29 @@ private async Task HandleErrors(string documentId, Exception e)
             await ThrowOnUnavailableStream(documentId, e).ConfigureAwait(false);
         }
 
-        private IDisposable ConcurrencyCheck()
+        private IDisposable _streamLockDisposable;
+        private async ValueTask<ReleaseStream> ConcurrencyCheckAsync()
```

Comment:
This isn't required.
The idea here is that the _external_ calls are not supposed to be concurrent, not internal ones
Those are already protected